src/lib/arch/riscv/Instruction.cc

```diff
@@ -14,14 +14,20 @@ const Register Instruction::ZERO_REGISTER = {RegisterType::GENERAL, 0};
 
 Instruction::Instruction(const Architecture& architecture,
                          const InstructionMetadata& metadata)
-    : architecture_(architecture), metadata(metadata) {
+    : architecture_(architecture),
+      metadata(metadata),
+      exception_(metadata.metadataException_) {
+  exceptionEncountered_ = metadata.metadataExceptionEncountered_;
   decode();
 }
 
 Instruction::Instruction(const Architecture& architecture,
                          const InstructionMetadata& metadata, uint8_t latency,
                          uint8_t stallCycles)
-    : architecture_(architecture), metadata(metadata) {
+    : architecture_(architecture),
+      metadata(metadata),
+      exception_(metadata.metadataException_) {
+  exceptionEncountered_ = metadata.metadataExceptionEncountered_;
```

Comment:
If the constructors are un-used then I would vote to remove them.

In AArch64 there is also the concept of not yet implemented alias'. It may be worth checking out how that is implemented to be consistent across the project, and will prevent the need for a "metadata exception"

Reply:
Replying to [this comment](https://github.com/UoB-HPC/SimEng/pull/303#issuecomment-1838937835) : 

I see. More information being printed for NYI alias' would be beneficial. @jj16791 your thoughts on this?

Reply:
I can update the AArch64 implementation to follow this if we think this is a good idea

Reply:
Should I copy over the functionality? And instead of reusing the ```ExecutionNotYetImplemented``` exception should I add a new ```AliasNotImplemented``` exception to provide more precise detail?
@jj16791 @FinnWilkinson

Reply:
Implemented. As for removing unused constructors this is also the case for the AArch64 Instruction constructor which takes an exception. Should this be removed also @FinnWilkinson @jj16791

Reply:
The second constructor which takes an exception is needed for things like:
```
    uop = std::make_shared<Instruction>(*this, metadataCache.front(),
                                        InstructionException::MisalignedPC);
```
so shouldn't be removed

Reply:
Ah yes, this is the case in RISC-V also. Seems like IDEs have trouble finding this usage

Reply:
I can't find a usage of the second RISC-V constructor so removed